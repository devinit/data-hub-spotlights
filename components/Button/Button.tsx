import React, { FunctionComponent } from 'react';

interface ButtonProps extends React.HTMLProps<HTMLButtonElement> {
  className?: string;
  onClick?: () => void;
}

const Button: FunctionComponent<ButtonProps> = ({ children, className, onClick }) => {
  return (
    <button type="button" className={className} onClick={onClick}>
      {children}
    </button>
  );
};
export { Button };
