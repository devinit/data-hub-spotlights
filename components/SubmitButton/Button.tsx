import classNames from 'classnames';
import React, { FunctionComponent } from 'react';

interface ButtonProps extends React.HTMLProps<HTMLButtonElement> {
  className: string;
}

const Button: FunctionComponent<ButtonProps> = ({ children, className }) => {
  return (
    <button type="button" className={classNames('button', className)}>
      {children}
    </button>
  );
};
export { Button };
