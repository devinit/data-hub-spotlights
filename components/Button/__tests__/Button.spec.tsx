/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { Button } from '../Button';

describe('Footer', () => {
  test('renders correctly', () => {
    const renderer = TestRenderer.create(<Button>Caption</Button>).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
