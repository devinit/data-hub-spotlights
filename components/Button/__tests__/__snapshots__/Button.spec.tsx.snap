// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  type="button"
>
  <style
    jsx={true}
  >
    
        display: block;
      
  </style>
  Caption
</button>
`;
